cmd/gateway-main.go

```diff
@@ -365,8 +365,10 @@ func gatewayMain(ctx *cli.Context, backendType gatewayBackend) {
 		// routes them accordingly. Client receives a HTTP error for
 		// invalid/unsupported signatures.
 		setAuthHandler,
-		// Add new handlers here.
+		// Add HTTP logging handler,
+		setLoggingHandler,
```

Comment:
> We shouldn't even add this handler if ENV is not enabled.

If ENV is not enabled, this line doesn't have any effect actually since setLoggingHandler won't create any new middleware. But you are right, it is probably more clear to check ENV here in this place.